exonum/src/runtime/rust/mod.rs

```diff
@@ -489,13 +493,22 @@ impl Runtime for RustRuntime {
         catch_panic(|| service.initialize(context, parameters))
     }
 
-    fn commit_service(
+    fn commit_service_status(
         &mut self,
         _snapshot: &dyn Snapshot,
         spec: &InstanceSpec,
+        status: InstanceStatus,
     ) -> Result<(), ExecutionError> {
-        let instance = self.new_service(spec)?;
-        self.add_started_service(instance);
+        match status {
+            InstanceStatus::Active => {
+                let instance = self.new_service(spec)?;
+                self.add_started_service(instance);
+            }
+
+            InstanceStatus::Stopped => {
+                self.remove_started_service(spec);
+            }
+        }
         self.new_services_since_last_block = true;
```

Comment:
Nit: This variable should probably be renamed for clarity to something like `service_changed_since_last_block`. I initially thought that you've forgotten to switch off HTTP API of stopped services because of it 🙃